test/common/formatter/substitution_formatter_test.cc

```diff
@@ -2401,8 +2402,83 @@ TEST(SubstitutionFormatterTest, GrpcStatusFormatterTest) {
   }
 }
 
-TEST(SubstitutionFormatterTest, GrpcStatusNumberFormatterTest) {
-  GrpcStatusFormatter formatter("grpc-status", "", absl::optional<size_t>(), true);
+TEST(SubstitutionFormatterTest, GrpcStatusFormatterSnakeStringTest) {
+  GrpcStatusFormatter formatter("grpc-status", "", absl::optional<size_t>(),
+                                GrpcStatusFormatter::Format::SnakeString);
+  NiceMock<StreamInfo::MockStreamInfo> stream_info;
+  Http::TestRequestHeaderMapImpl request_header;
+  Http::TestResponseHeaderMapImpl response_header;
+  Http::TestResponseTrailerMapImpl response_trailer;
+  std::string body;
+
+  std::array<std::string, 17> grpc_statuses{"OK",
+                                            "CANCELLED",
+                                            "UNKNOWN",
+                                            "INVALID_ARGUMENT",
+                                            "DEADLINE_EXCEEDED",
+                                            "NOT_FOUND",
+                                            "ALREADY_EXISTS",
+                                            "PERMISSION_DENIED",
+                                            "RESOURCE_EXHAUSTED",
+                                            "FAILED_PRECONDITION",
+                                            "ABORTED",
+                                            "OUT_OF_RANGE",
+                                            "UNIMPLEMENTED",
+                                            "INTERNAL",
+                                            "UNAVAILABLE",
+                                            "DATA_LOSS",
+                                            "UNAUTHENTICATED"};
+  for (size_t i = 0; i < grpc_statuses.size(); ++i) {
+    response_trailer = Http::TestResponseTrailerMapImpl{{"grpc-status", std::to_string(i)}};
+    EXPECT_EQ(grpc_statuses[i], formatter.format(request_header, response_header, response_trailer,
+                                                 stream_info, body));
+    EXPECT_THAT(
+        formatter.formatValue(request_header, response_header, response_trailer, stream_info, body),
+        ProtoEq(ValueUtil::stringValue(grpc_statuses[i])));
+  }
+  {
```

Comment:
Parenthesis not required here.

Reply:
I agree it is not required but it is universally used in this test. I guess it makes each sub test cases more clear so I prefer to keep it.